### ISSUE TITLE
Nested generic parameter types don't get imported

### ISSUE THREAD

**Issue by ResoDev** [Milad-Akarie/injectable#21]
A class with the following constructor
```dart
@lazySingleton
@RegisterAs(LocalForecastCache)
class HiveForecastCache implements LocalForecastCache {
  final Box<ForecastDto> weatherForecastBox;

  HiveForecastCache(this.weatherForecastBox);
}
```

results in the following generated code.
```dart
  g.registerLazySingleton<LocalForecastCache>(() => HiveForecastCache(
        g<Box<ForecastDto>>(),
      ));
```

However, the nested `ForecastDto` generic parameter isn't automatically imported to the generated file.

**Comment by Milad-Akarie** [comment]
ya it was reported. I'm working on it along with support for get_it 4.0.0

**Comment by ResoDev** [comment]
Thanks, I'll just notify you here that exactly the same thing happens in **auto_route**.

**Comment by Milad-Akarie** [comment]
@ResoDev ya I Just noticed when I copied the code over that I screwed it up in my last update lol

**Comment by Milad-Akarie** [comment]
@ResoDev it's been fixed in injectable_generator: 0.2.4, give it a try.
